internal/hardware/validator.go

```diff
@@ -130,6 +131,47 @@ func (v *validator) GetHostRequirements(role models.HostRole) models.HostRequire
 	}
 }
 
-func IsVirtSupported(inventory *models.Inventory) bool {
-	return funk.Contains(inventory.CPU.Flags, intelVirtCpuFlag) || funk.Contains(inventory.CPU.Flags, amdVirtCpuFlag)
+func (v *validator) GetClusterHostRequirements(ctx context.Context, cluster *common.Cluster, host *models.Host) (*models.ClusterHostRequirements, error) {
+	operatorsRequirements, err := v.operatorsAPI.GetRequirementsBreakdownForRole(ctx, cluster, host.Role)
+	if err != nil {
+		return nil, err
+	}
+
+	hostRequirements := v.GetHostRequirements(host.Role)
+	ocpRequirements := models.ClusterHostRequirementsDetails{
+		CPUCores:                         hostRequirements.CPUCores,
+		RAMMib:                           conversions.GibToMib(hostRequirements.RAMGib),
+		DiskSizeGb:                       hostRequirements.DiskSizeGb,
+		InstallationDiskSpeedThresholdMs: hostRequirements.InstallationDiskSpeedThresholdMs,
+	}
+	total := totalizeRequirements(ocpRequirements, operatorsRequirements)
```

Comment:
you could pass `ocpRequirements` by pointer to avoid copying the contents as argument.
```go
total := totalizeRequirements(&ocpRequirements, operatorsRequirements)
```

Reply:
The guidelines for small structs are to use pass-by-value and if pointer is not needed in the method, than probably it shouldn't be a pointer argument (https://github.com/golang/go/wiki/CodeReviewComments#pass-values).